### ISSUE TITLE
List of minor bugs, missings or inexplicit behavior

### ISSUE THREAD

**Issue by TomMuc1** [Ysurac/FlightAirMap#341]
map side-bar menu: 'Display flight remaining route on click' and 'Display flight route on click' not deselectable

**Comment by TomMuc1** [comment]
found no setting to just display just ALL aircrafts that my site receives messages from AT THE MOMENT

**Comment by TomMuc1** [comment]
while other map interfaces show instantly all aircrafts i receive - flightairmap needs often several minutes to do so

**Comment by TomMuc1** [comment]
top menubar -> explore -> latest accident/incident
while all other within explore are very fast and responsive these two menu-items take very long, freeze the site and put huge cpu-load (mysql) on the machine

**Comment by TomMuc1** [comment]
a options/settings html-page like virtualradarserver has would be a very nice enhancement and/or better documentation of require/settings.php

**Comment by TomMuc1** [comment]
one feature of virtualradarserver i love - and did not find (maybe overlooked) is the option to do range-plots for my site even over a long time period and different altitudes

**Comment by TomMuc1** [comment]
at least at my site (1.500msg/sec) very high cpu-load (70-100%) while running daemon-spotter.php 24/7 via cronjob and using port 30003 on dump1090-mutability as data-source.

**Comment by TomMuc1** [comment]
altitude colors and aircraft outlines of this app are maybe a little nicer - just my single opinion: http://radarcape-demo.jetvision.de/gmap.html

**Comment by Ysurac** [comment]
* map side-menu: fixed, I will push on git soon
* all aircrafts are always displayed
* without using something like websocket that push data to map, it's not possible. Many map work the same way as flightairmap: a refresh. **$globalMapRefresh** set when there is data refresh.
* I will try to optimize that, many checks are done here. It's fast with postgreSQL but slow with MySQL
* It's why **install/index.php** is for. Even if there is many options and this would need to be reorganized
* I never used VRS so I don't know what this look like
* daemon-spotter.php using cronjob for a SBS source is bad, doesn't work well and is CPU intensive. Using conjob a lot of check/features/optimization are not available and this may write more data to DB

**Comment by TomMuc1** [comment]
hahaha - lightningfast as always :) never saw an app where even these minor bugs were fixed soo fast! thanx

**Comment by Ysurac** [comment]
* I don't like outlines
* You are speaking about "aircrafts color based on altitude" ?

**Comment by TomMuc1** [comment]
yep - the altitude color - but again just my taste maybe other see this different ...

**Comment by TomMuc1** [comment]
in this post you can see two of these range plots: https://discussions.flightaware.com/t/post-your-piaware-setup/17467/158?u=tommuc

**Comment by TomMuc1** [comment]
> I will try to optimize that, many checks are done here. It's fast with postgreSQL but slow with MySQL

i'll have a look into table structure and indexes of mysql - maybe i can find out what the problem could be with mysql ...

**Comment by TomMuc1** [comment]
and please let me know when you managed to add aircraft.json and lighttpd support - but again - no hurry - i'm just curious :)

**Comment by Ysurac** [comment]
There is a (dirty) polar in FlightAirMap when stats are enable for the source (**sourcestats** should be TRUE for the source). I don't remember if this work when using spotter-daemon.php as cronjob, maybe not.

Maybe check are done when loading page to find if a aircraft that crash was not seen before. I will try to make this another way. MySQL is always slower than postgreSQL for FlightAirMap usage.

aircraft.json support is added in latest commit. (even if daemon-spotter.php running as daemon with TCP source is better for data). As cronjob this will use less CPU, but don't run it as cronjob when you can.

**Comment by TomMuc1** [comment]
missing option to deselect an aircraft on map without selecting another - again - maybe i overlooked something

**Comment by Ysurac** [comment]
There is a little X on the upper right corner of the flight info box, this will close info box and deselect aircraft.

**Comment by TomMuc1** [comment]
> aircraft.json support is added in latest commit. (even if daemon-spotter.php running as daemon with TCP source is better for data). As cronjob this will use less CPU, but don't run it as cronjob when you can.

unbelivable!!! what is the best way to update the app to new commits? and what is the best/easiest way to switch from 30003 input to aircraft.json without running the installer again?

**Comment by TomMuc1** [comment]
> There is a little X on the upper right corner of the flight info box, this will close info box and deselect aircraft.

aaah - ok - silly me :)

**Comment by Ysurac** [comment]
**git pull**
You can run the installer, when initial install is done it only update table/options when needed. It should be run after each update.

**Comment by TomMuc1** [comment]
hmmm - that's what i did - but:

pi@RaspPiADSB5:/var/www/html $ ls
flightairmap  index.html  index.lighttpd.html  phpinfo.php
pi@RaspPiADSB5:/var/www/html $ git pull
fatal: Not a git repository (or any of the parent directories): .git
pi@RaspPiADSB5:/var/www/html $

**Comment by Ysurac** [comment]
In flightairmap directory and if you get it via git the first time.

**Comment by TomMuc1** [comment]
pi@RaspPiADSB5:/var/www/html/flightairmap $ git pull
remote: Counting objects: 10, done.
remote: Compressing objects: 100% (1/1), done.
remote: Total 10 (delta 9), reused 10 (delta 9), pack-reused 0
Unpacking objects: 100% (10/10), done.
From http://github.com/ysurac/flightairmap
   87b9d11..9e0f727  master     -> origin/master
Updating 87b9d11..9e0f727
error: Your local changes to the following files would be overwritten by merge:
        index.php
        install/index.php
        polar-geojson.php
        require/class.SpotterImport.php
Please commit your changes or stash them before you merge.
Aborting

**Comment by TomMuc1** [comment]
i did zero changes to the files named above beside setting them to 777 while all the install hassle was some days ago ...

**Comment by Ysurac** [comment]
It's a change.  2 choices: 
* Delete this files, and do git pull again
or do that:

      git fetch --all
      git reset --hard origin/master
      git pull origin master

**Comment by TomMuc1** [comment]
ok - worked instantly. but now did a reboot and in /install Data source can't see the option for aircraft.json ...

**Comment by TomMuc1** [comment]
ok - the index.php i have in install directory is different from the one that is in master on github ...

**Comment by TomMuc1** [comment]
that means the git pull did not what was supposed to do

**Comment by TomMuc1** [comment]
delete all files and clone completely from master?

**Comment by Ysurac** [comment]
yes maybe better, you can keep the require/settings.php

**Comment by TomMuc1** [comment]
pi@RaspPiADSB5:/var/www/html $ sudo rm -r flightairmap/
pi@RaspPiADSB5:/var/www/html $ sudo git clone --depth=1 http://github.com/ysurac/flightairmap
Cloning into 'flightairmap'...
remote: Counting objects: 2943, done.
remote: Compressing objects: 100% (2788/2788), done.
remote: Total 2943 (delta 252), reused 2564 (delta 131), pack-reused 0
Receiving objects: 100% (2943/2943), 91.11 MiB | 4.42 MiB/s, done.
Resolving deltas: 100% (252/252), done.
Checking out files: 100% (2959/2959), done.
pi@RaspPiADSB5:/var/www/html $ sudo chmod -R 777 flightairmap/

now have the actual install/index.php - but agian see no option for aircraft.json

**Comment by Ysurac** [comment]
why do you chmod 777 ?
It's *Dump1090 aircraft.json*

**Comment by TomMuc1** [comment]
> why do you chmod 777 ?

because i was too lazy to see which files needs what permissions

i find 'aircraft.json' in the source of install/index.php but not on html output in chrome!

**Comment by TomMuc1** [comment]
cleard browser cache and restarted browser - now in option menu

**Comment by Ysurac** [comment]
When you run install/index.php it say which file need write permission.

In source you can choose the format, even if using **auto** it should detect it.
If the file is locally available use file:///run/idontknowwhereitisforyou/aircraft.json else if over webserver http://ip/../aircraft.json

**Comment by Ysurac** [comment]
ok, never add this problem. In latest commit this will not be cached by browser.

**Comment by TomMuc1** [comment]
no wonder with all this back and force I MADE with the browser :)

**Comment by TomMuc1** [comment]
output from terminal:

New flight...
*********** New aircraft hex : 4B0F76 ***********
New flight...
*********** New aircraft hex : 0200AF ***********
New flight...
*********** New aircraft hex : 3D3719 ***********
New flight...
*********** New aircraft hex : 3D16CD ***********
Sleeping...
New flight...
*********** New aircraft hex : 4CA6B6 ***********
New flight...
*********** New aircraft hex : 4CA948 ***********
New flight...
*********** New aircraft hex : 3C6608 ***********
New flight...
*********** New aircraft hex : 3C4891 ***********
New flight...
*********** New aircraft hex : 3C6445 ***********
New flight...
*********** New aircraft hex : 501D1E ***********
!!! Too much distance in short time... for
Time : 52s - Distance : 22601m -Speed : 1564.6846153846 km/h - Lat : 47.345352 - long : 10.156517 - prev lat : 47.472157 - prev long : 9.921781
ø Calculated Heading for 501E1B : 359
!!! Too much distance in short time... for
Time : 52s - Distance : 26108m -Speed : 1807.4769230769 km/h - Lat : 48.051918 - long : 13.183242 - prev lat : 48.252512 - prev long : 13.000296
Sleeping...

**Comment by TomMuc1** [comment]
bot no aircrafts show up -> made a second source via install (the old working 30003) but where can i switch between sources in flightairmap html interface?

**Comment by TomMuc1** [comment]
if i delete aircraft.json from sources it autoreverts instantly to 30003 and the planes show up.
what i wonder in addition is that daemon-spotter.php sleeps most of the time when aircraft.json is set up as source. but as the aircraft.json is refreshed every second it should pick up a new every second! what and where did you setup the interval to fetch the new file again and again?

**Comment by Ysurac** [comment]
aircraft.json support fixed in latest commit.

Source are agregated, but http and TCP source can't be used at the same time.
 You can decrease the sleeping time using **$globalMinFetch** (I forget to add it to install/index.php...)

**Comment by TomMuc1** [comment]
the git pull again did not work

**BUT**

**i cant be happier!** set $globalMinFetch = '1' (forwhatever the singlequotes are good for with numbers:))) and now lots of aircrafts instantly show up on the map **INCLUSIVE ALL (WITH ANONYMIZED) MLAT** **AND** as i always assumed **CPU-LOAD** of daemon-spotter.php now is between **5-10%** instead of before **70-100%**

**PERFECT!!! THANX SO MUCH :)**

**Comment by Ysurac** [comment]
* With the chmod -R 777, git pull will always fails.

* It's because you use it as cron job. It's not really usable as cron for any TCP source. With cron I need to do all checks everytime the cron run.

**Comment by TomMuc1** [comment]
- ok did not know that github pull has problems when all files and dirs are rwx from all
- with the cron vs system-service i'm not really convinced that this is the real problem. of course cron does not these system optimization-things that system-service can - your script isn't called fresh every second but simply loops forever until machine is restarted. i'd still say all this message fetching and combining is muuuch more cpu-intensive than just reading the json every second. and i'm not sure if you ever tried this on a raspi with a message-rate i have ...

anyways - i'm really happy that you gave the aircraft.json a try! and we'll see if over time you start to love this option too :) now there is no more downside to use your app even on a cheap old raspberry pi2 and when it will be possible to adapt the htaccess to lighttpd i'm sure nearly every feeder from flightaware want's to run your app!

**Comment by Ysurac** [comment]
Cron job use less CPU but write and read a lot more data to/from DB, less checks possibles, speed and heading can't be calculated if not available,...
But it's your choice, maybe better for a slow Raspberry PI (at least if DB is not on it, else maybe not).

I'm trying to do the change for lighttpd.

**Comment by TomMuc1** [comment]
ok i'll give system-service a try as you did with .json :)

- what exactly are the things that do not work when running daemon-spotter.php from cronß and why?

- what are the downsides of aircraft.json you still see vs port 30003 input?

**Comment by Ysurac** [comment]
* From cron all check that need previous data doesn't work because I can't put data in memory via cron job. So no speed/heading calculation or removing of false data, to only put data in DB for history track when needed (when data not on the same line, when data didn't change a lot,...). There is also more read of DB (I can't put some data in memory) and more write (I don't know if data really change since last check).
* TCP is realtime, don't need to write a file (even if it's in memory), easier to access from another computer, it's a standard format used by a lot of devices,...

**Comment by Ysurac** [comment]
lighttpd configuration is added in install directory.
I will have to write a little tutorial for that.

**Comment by TomMuc1** [comment]
ok - got your points :)

what i'll do is setup a systemctl service instead of cron ...

**Comment by TomMuc1** [comment]
you make me fear - are you human? - every single wish/idea i had - some minutes later your'e done with the job. how many arm, heads and seconds within an hour you guys over there in france have???

**Comment by TomMuc1** [comment]
btw. preventing browser from caching often it is more reliable to add http headers in addition to html headers - because the http headers can depend on global php setup ... e.g. this way:
<?php
header("Cache-Control: no-store, no-cache, must-revalidate, max-age=0");
header("Cache-Control: post-check=0, pre-check=0", false);
header("Pragma: no-cache");
?>

**Comment by Ysurac** [comment]
This doesn't really change anything if there is no proxy involved but why not.

**Comment by TomMuc1** [comment]
:))) this totally bavarian style of humour

**Comment by TomMuc1** [comment]
just one last question today. i saw you did something with the .htaccess for lighttpd - is there something i have to be aware of when making a test install of flightairmap on a dump1090 install with lighttpd - or should i wait some days until you finished this all?

**Comment by Ysurac** [comment]
Read this dirty wiki doc: https://github.com/Ysurac/FlightAirMap/wiki/Lighttpd

**Comment by TomMuc1** [comment]
thanx - have a nice evening!

**Comment by Ysurac** [comment]
nice evening too.

**Comment by TomMuc1** [comment]
side-bar-menu:
- Display flight remaining route on click - now uncheck works but does not apply to map
- Display flight info as popup does not apply to map

install:
- ground station setting 'altitude' does not say meters or feet input needed

**Comment by Ysurac** [comment]
sidebar menu:
* Fixed
* Already work, it's the red line that display when I have airports info
* Disabled now.
* Should work on a tablet with GPS. Need to check again...
* Fixed
* If airspaces data are in DB, this should work. Setting are not saved but it's not a bug.
* Same. I don't know if settings like that should be saved, this display lot of data on client browser and can make problems on slow computer.

top menu:
* I don't know what you are meaning. aircraft icon is not a real menu, only a shortcut.

install:
* altitude is displayed as set. I added in settings that this should be meters, but for now this will not change anything.

**Comment by TomMuc1** [comment]
- Plot your Location nothing happens on click
on my ipad with gps it now has an effect - but a bizarre one - the whole map turns 360 degree depending on the direction the ipad looks to

- aircraft icon (top left) 
what i meant was that all spaces between menu items (explore, search, statistice etc) are bigger than the spacing between the aircraft-icon and the menu-item 'explore'
so i'd say simply shift the aircraft icon a little to left then i is placed middle above the menu-bar and has same spacing to 'explore'

made a quick photoshop mockup and sent to contact@flightairmap.fr

**Comment by TomMuc1** [comment]
haha - thanks :) don't you think it looks better now - top-menu and side-bar-menu?

**Comment by Ysurac** [comment]
True, it's better. It was a bug in my CSS.

**Comment by TomMuc1** [comment]
where do i find the lighttpd config file?

**Comment by Ysurac** [comment]
In *install/flightairmap-lighttpd-conf.include*.
Read https://github.com/Ysurac/FlightAirMap/wiki/Lighttpd

**Comment by TomMuc1** [comment]
i've read this - not the .include but the flightairmap.conf file itself is what i'm looking for

**Comment by Ysurac** [comment]
Lighttpd configuration is customized by Debian. I will add a Debian conf in tutorial.

**Comment by TomMuc1** [comment]
the file that then resides in /etc/lighttpd/conf-availible/ to configure lighttpd via
`sudo lighty-enable-mod flightairmap`

**Comment by Ysurac** [comment]
Yes, it's only for debian/raspbian. The conf I made is for all distribution. I will make a conf for Debian/Raspbian.

**Comment by TomMuc1** [comment]
yep as this will be the most wanted version because lighttpd in my opinion makes only sense when flightairmaps installs on raspi with dump1090 that already uses lighttpd

**Comment by Ysurac** [comment]
Dump1090 also provide nginx conf, and it's far better.

**Comment by TomMuc1** [comment]
i know and you are fully right! but at the moment about 95-100% of the feeders run pi with dump1090 and lighttpd. and it's all about making it as easy as possible for the majority to use your application. just flightaware has more than 10,000 feeders

**Comment by Ysurac** [comment]
It's available in latest commit and doc is updated: https://github.com/Ysurac/FlightAirMap/wiki/Lighttpd
Should work.

**Comment by TomMuc1** [comment]
the standard path for lighttpd is /var/www/html - i changed this - but:

`pi@RaspPiADSB0:~ $ lighty-enable-mod 89-flightairmap-lighttpd.conf
Ignoring unknown module: 89-flightairmap-lighttpd.conf
Run "service lighttpd force-reload" to enable changes`

**Comment by TomMuc1** [comment]
aaah - problem is - when enabling module the '.conf' extension is not recognized.
this way it worked:

	sudo apt-get install php5-cgi lighttpd
	sudo cp /var/www/html/flightairmap/install/89-flightairmap-lighttpd.conf /etc/lighttpd/conf-available
	sudo lighty-enable-mod 89-flightairmap-lighttpd
	sudo /etc/init.d/lighttpd force-reload

**Comment by Ysurac** [comment]
Sorry, I updated the doc.

**Comment by TomMuc1** [comment]
you don't have to say sorry - when you realize each and every wish from a single user within minutes. and moreover we all use your work for FREE!!! so i have to say a big THANK YOU for your help and patience :)

cheers
tom

**Comment by TomMuc1** [comment]
p.s. in addition i'll ask the developer of dump1090-mutability why lighttpd is used to be the standard and not nginx - and what his future plan is ...

**Comment by TomMuc1** [comment]
haha - and btw. this time i set up the mysql not on the raspi i run flightairmaps but on a 4-core i7 mac-mini server with 16gb ram. now everything is lightningfast :)

**Comment by TomMuc1** [comment]
an issue i was runnuning into again and again - see example above - and now when activating the system-service for daemon-spotter.php is the path problem. in so many files across the app /var/www/flightairmap/... is hardcoded into the files. how about having this path in one distinct variable so if one has to change to e.g. /var/www/html/flightairmap/... all other files pick up the path from there?

**Comment by TomMuc1** [comment]
in addition here was 'Wep_App' in the path too - never saw this directory anywhere ???

**Comment by TomMuc1** [comment]
see flightairmap/install/init/flightairmap.service.systemd

	[Unit]
	Description=FlightAirMap
	Requires=mysql.service
	After=mysql.service

	[Service]
	ExecStart=/var/www/flightairmap/Web_App/scripts/daemon-spotter.php
	Restart=always
	RestartSec=10

	[Install]
	WantedBy=multi-user.target

**Comment by TomMuc1** [comment]
a lighttpd problem i've found is with link from top menu explore->latest accident/incident all browsers say 'too much redirects' instead of showing the page

**Comment by Ysurac** [comment]
Web_App removed in latest commit.
No real other choices than hardcoding path in init, if I put a file with the variable I will have to hardcode the path to this file...

**Comment by TomMuc1** [comment]
sorry i was unclear :) i thought the user inputs his path in the install script -> then the installer generates or rewrites the files in /flightairmaps

**Comment by TomMuc1** [comment]
GEOID HEIGHT CALCULATOR:

    Exception: Can't open /var/www/html/flightairmap/require/../data/egm96-15.pgm in /var/www/html/flightairmap/require/class.GeoidHeight.php:47 Stack trace: #0 /var/www/html/flightairmap/tools-geoid.php(6): GeoidHeight->__construct() #1 {main}

**Comment by Ysurac** [comment]
* I'm trying to fix lighttpd problem. Seems a strange bug...
* I will not use install script to change path in config files, this will need far too much write rights.
* You need to run *update_db.php* after install, this should download the needed file.

**Comment by TomMuc1** [comment]
- sorry that i started this lighttpd thing ...
- understand
- update_db.php is running since about 30-60 minutes - how long does this script run average?

**Comment by Ysurac** [comment]
* fixed

* Can be very slow for first run when adding accident and incidents (I need to find a way to make this faster). It also download all 3d models if enabled.

**Comment by TomMuc1** [comment]
git pull-> lighttpd restart->WORKS LIKE A CHARM :)
you are unbelivable!!!

**Comment by TomMuc1** [comment]
what i wanted to ask is if there maybe in future could be a way to display whether or not position/altitude data was computed with mlat by displaying mlat in the aircraft data popup-window

**Comment by TomMuc1** [comment]
posted some pix - as i'm so happy with the new tool i have now :)
https://discussions.flightaware.com/t/one-hell-of-a-ride-flightairmaps-the-way-better-virtualradarserver/26122?u=tommuc
https://discussions.flightaware.com/t/one-hell-of-a-ride-flightairmaps-the-way-better-virtualradarserver/26122/41?u=tommuc

**Comment by Ysurac** [comment]
MLAT flag is only in dump1090.json format, putting this in a new *source name* this may be possible but not really accurate because I don't save every coordinates, I can merge several sources,...

I redirected MLAT to another port and use that as another source, I should put the config in the doc.

**Comment by TomMuc1** [comment]
just show up in the aircraft popup on map-window wre great - no need to flag this in database data

**Comment by TomMuc1** [comment]
you will see over time that most users will use the aircraft.json input as it already hold all mlat without loop-back prone port 30003 mlat tinkering. in addition there is no way to get the anonymized mlat from flightaware to port 30003. moreover a much better source than basestation is the beast-binary format but again more cpu-intensive to process with php ...

**Comment by Ysurac** [comment]
beast binary doesn't really change anything over basestation format.
I will make some test for mlat, I didn't play with this since a long time...

**Comment by TomMuc1** [comment]
> beast binary doesn't really change anything over basestation format.

starts with that you can determine whether or not mlat :)

what is a major problem with basestation-format as fa, fr24, planefinder etc. are not able to detect if someone by accident loops them mlat data back ...

**Comment by Ysurac** [comment]
* Maybe but I don't know where it is in beast format.
* True for basestation format.

**Comment by TomMuc1** [comment]
to be honest i do not know either :))) never looked into the data - i was just the parrot who repeated what i once read from mutability developer. but it must be there as is combine adsb/mlat on one pi - then fetch via socat from another pi the combined beast-stream and then tis pi produces the aircraft.json that again clearly differentiates in subkey 'mlat' of each aircraft.json row ...

**Comment by Ysurac** [comment]
I looked in dump1090, and found this: *#define MAGIC_MLAT_TIMESTAMP 0xFF004D4C4154ULL*, so it is using a special timestamp.

**Comment by TomMuc1** [comment]
you are unbelievable and lightning-fast :) and what i unsertood from what i once read - doing things like this with basestation format is hard or impossible e.g. flightawares ~at the beginning of transponder hex code of their anonymized mlat feedback

**Comment by TomMuc1** [comment]
in addition i think the basestation hardware has no relevance today so why use their protocol. opposite to that the jetvision beast and radarcape are still the top of the line receivers ...

**Comment by Ysurac** [comment]
because it's the only TCP not binary format widely available.

**Comment by TomMuc1** [comment]
yeeeeeahhh - and i know you do not like 'my' .json format too :)))

**Comment by Ysurac** [comment]
I added in the doc what I use to get MLAT info using TCP: https://github.com/Ysurac/FlightAirMap/wiki/ADS-B

I will add MLAT for dump1090.json asap.

**Comment by TomMuc1** [comment]
no no - you have to take a rest over the weekend! to be fresh and in good mood - when i'll come around the corner with my next silly idea for improvement :)

**Comment by Ysurac** [comment]
MLAT for aircraft.json added in latest. Should work.

**Comment by TomMuc1** [comment]
update done - but how/where can i see on map whether plane is mlat or not?

**Comment by Ysurac** [comment]
When clicking on aircraft this should display MLAT as source name.
If not I will have to make some test ;)

**Comment by TomMuc1** [comment]
easy possible that i just overlook this - where exactly mlat should be displayed?

**Comment by Ysurac** [comment]
When you click, under **SOURCE**. But I don't have any mlat in my aircraft.json for now, so I can't test.

**Comment by TomMuc1** [comment]
to send you a screenshot what email i have to use?

**Comment by Ysurac** [comment]
support@flightairmap.com

Do you have anything in aircraft.json in *"mlat":*, I always have [], strange. Will look at this another day.

**Comment by TomMuc1** [comment]
you have mail :)

**Comment by Ysurac** [comment]
You can also put screenshot directly here.

There is no MLAT info in the screenshot. It should display at down right of flight info if it's MLAT (else nothing).

**Comment by daviessm** [comment]
Can confirm MLAT works here using the instructions above.

![image](https://user-images.githubusercontent.com/3835494/31032983-50c7aaee-a556-11e7-8760-bcade34a87a6.png)

**Comment by TomMuc1** [comment]
what is your source 30003 or .json

**Comment by daviessm** [comment]
See my edit - port 30106 TCP connection to piaware.

**Comment by TomMuc1** [comment]
so - maybe this is the reason i do not see mlat announced in the box - my input is aircraft.json

**Comment by Ysurac** [comment]
TCP is always better :)
For aircraft.json this should work if there is data in *mlat* field. I have none for unknown reason.

**Comment by TomMuc1** [comment]
hahahaha :)))

nope - there is never real data in field mlat see here:

aircraft without mlat:

    ,"mlat":[]

aircraft with mlat:

    "mlat":["lat","lon","track","speed","vert_rate"]

**Comment by TomMuc1** [comment]
i sent you a zipped aircraft.json that has mlat aircrafts inside. so you could see what your code for json has to check for ...

**Comment by Ysurac** [comment]
I know...
I tested with a aircraft.json feed with working MLAT and it's working .

**Comment by TomMuc1** [comment]
hmmm - so then i'm the problem here :) i have to look into that ...

**Comment by Ysurac** [comment]
Did you restart *spotter-daemon.php* ?

**Comment by TomMuc1** [comment]
nope just lighttpd :(

**Comment by TomMuc1** [comment]
sorry for wasting your time! my fault i forgot to restart the spotter service. now works instantly fine and i see always my source json and if mlat it says json_mlat :) thanx a lot again :)

**Comment by Ysurac** [comment]
No problem.
Now I'm out for the week end ;)

**Comment by TomMuc1** [comment]
have a nice weekend! you deserved this :)

greetings from bavaria/germany
tom

**Comment by TomMuc1** [comment]
nice - the guys from flightaware forum start more and more to run your application. in addition i sent the owner/developer of radarcape some infos about the app ...

**Comment by TomMuc1** [comment]
3d - at least after last commit/pull doesn't work anymore:

`Error constructing CesiumWidget.
Visit http://get.webgl.org to verify that your web browser and hardware support WebGL.  Consider trying a different web browser or updating your video drivers.  Detailed error information is below:

RuntimeError: The browser supports WebGL, but initialization failed.
Error
    at new t (https://cesiumjs.org/releases/1.37/Build/Cesium/Cesium.js:486:4061)
    at new N (https://cesiumjs.org/releases/1.37/Build/Cesium/Cesium.js:535:8529)
    at new Ee (https://cesiumjs.org/releases/1.37/Build/Cesium/Cesium.js:552:13327)
    at new A (https://cesiumjs.org/releases/1.37/Build/Cesium/Cesium.js:565:25179)
    at new Z (https://cesiumjs.org/releases/1.37/Build/Cesium/Cesium.js:568:25781)
    at http://192.168.143.10/flightairmap/js/map.3d.js.php:281:14`

**Comment by Ysurac** [comment]
Working for me.
What do you have on line 281 of map.3d.js when you have that error ?

**Comment by TomMuc1** [comment]
ooops - thought you were out for the weekend:

var viewer = new Cesium.Viewer('live-map', {
    sceneMode : Cesium.SceneMode.SCENE3D,
    baseLayerPicker: false,
    imageryProvider : imProv,

**Comment by Ysurac** [comment]
Strange...
What **type of map** are you using ?

**Comment by TomMuc1** [comment]
openstreetmap or yandex - but you were right has nothing to do with latest commit.
had this error never before ...

**Comment by TomMuc1** [comment]
i'm not sure if i ever tried 3d mode on lighttpd before - my earlier installs based on apache never sowed this behavior. no idea whether this can correlate with lighttpd ...

**Comment by Ysurac** [comment]
It's not a problem with lighttpd, it's working for me.
Can you try https://cesiumjs.org/ click on **Tap to interact** and tell me if you can play with the world.

**Comment by TomMuc1** [comment]
yup - works instantly. but cmon today is sunday i do not want to disturb your evening. tomorrow i install apache again and see what happens - and try my other notebook. i'll guess i made a mistake with the new setup. this were all steps i made:


	sudo apt-get install php7.0-common php7.0-cgi php7.0-mysql php7.0-sqlite php7.0-curl php7.0 php7.0-xml php7.0-zip php7.0-gd php7.0-json
	cd /var/www/html
	sudo git clone --depth=1 http://github.com/ysurac/flightairmap
	sudo chmod 666 flightairmap/require/settings.php
	sudo chmod 777 flightairmap/install/tmp
	sudo chmod 666 flightairmap/data
	sudo chmod 666 flightairmap/images/airlines
	sudo cp /var/www/html/flightairmap/install/89-flightairmap-lighttpd.conf /etc/lighttpd/conf-available
	sudo nano /etc/lighttpd/conf-available/89-flightairmap-lighttpd.conf
	sudo lighty-enable-mod flightairmap-lighttpd
	sudo /etc/init.d/lighttpd force-reload
	sudo cp /var/www/html/flightairmap/install/init/flightairmap.service.systemd /etc/systemd/system/flightairmap.service
	sudo nano /etc/systemd/system/flightairmap.service
	sudo chmod 644 /etc/systemd/system/flightairmap.service
	sudo systemctl enable flightairmap.service

**Comment by TomMuc1** [comment]
hahaha - even stranger - works instantly on my new macbook pro i7 4gb graphic ram.
but all the days it was working on my old mac book pro too. i changed nothing there - and was using over the last days when i startted with flightairmap only this old notebook ...

**Comment by TomMuc1** [comment]
yes and yes :)

**Comment by TomMuc1** [comment]
restarted the old mac-book pro now it is again working :) i have one suspicion - all these high end level pro mac books have two graphic engines builtin. an intel within chipset and a second dedicated one. dependend on generation they switch in different ways between them. i think that for what ever reason something went there wrong. but what makes me wonder why did the link you sent me without problem and google earth pro too???

**Comment by TomMuc1** [comment]
i wrote this for a flightaware forum member to assist him with install - maybe you find this helpful too for your wiki anyways:

	passwd
	sudo raspi-config
	sudo apt-get update
	sudo apt-get upgrade
	sudo apt-get install php7.0-common php7.0-cgi php7.0-mysql php7.0-sqlite php7.0-curl php7.0
	sudo apt-get install mysql-server mysql-client
	sudo mysql -u root -p (leave password empty)
		update mysql.user set password=password('YOUR_DB_PASSWORD') where user='root';
		update mysql.user set plugin='' where user='root';
		flush privileges;
	sudo mysql -u root -p (use password you entered above)
		CREATE DATABASE 'adsb';
		ctrl + z
	sudo apt-get install build-essential debhelper librtlsdr-dev libusb-1.0-0-dev pkg-config tcl8.5-dev autoconf python3-dev python-virtualenv libz-dev git tclx8.4 tcllib tcl-tls itcl3 htop
	mkdir dump1090-mutability
	cd dump1090-mutability/
	git clone https://github.com/mutability/dump1090.git
	cd dump1090/
	dpkg-buildpackage -b
	cd ..
	sudo dpkg -i dump1090-mutability_1.15~dev_armhf.deb
	cd ~
	git clone https://github.com/flightaware/dump1090.git
	sudo cp -r dump1090/public_html  /usr/share/dump1090-mutability/public_html
	cd /usr/share/dump1090-mutability
	sudo mv html original-html
	sudo mv public_html html
	cd html
	sudo mv index.html gmap.html
	sudo wget -O  /etc/udev/rules.d/rtl-sdr.rules "https://raw.githubusercontent.com/osmocom/rtl-sdr/master/rtl-sdr.rules"
	sudo dpkg-reconfigure dump1090-mutability
	sudo shutdown -r now
	sudo apt-get install lighttpd
	sudo lighty-enable-mod dump1090
	sudo lighty-enable-mod fastcgi
	sudo lighty-enable-mod fastcgi-php
	sudo service lighttpd force-reload
	sudo apt-get install php7.0-xml php7.0-zip php7.0-gd php7.0-json
	sudo nano /var/www/html/phpinfo.php
	sudo nano /etc/php/7.0/cgi/php.ini
	sudo shutdown -r now
	cd /var/www/html
	sudo git clone --depth=1 http://github.com/ysurac/flightairmap
	sudo chmod 666 flightairmap/require/settings.php
	sudo chmod 777 flightairmap/install/tmp
	sudo chmod 666 flightairmap/data
	sudo chmod 666 flightairmap/images/airlines
	sudo cp /var/www/html/flightairmap/install/89-flightairmap-lighttpd.conf /etc/lighttpd/conf-available
	sudo nano /etc/lighttpd/conf-available/89-flightairmap-lighttpd.conf (alter pathes to match your path)
	sudo lighty-enable-mod flightairmap-lighttpd
	sudo /etc/init.d/lighttpd force-reload
	sudo shutdown -r now
	sudo cp /var/www/html/flightairmap/install/init/flightairmap.service.systemd /etc/systemd/system/flightairmap.service
	sudo nano /etc/systemd/system/flightairmap.service (alter pathes to match your path)
	sudo chmod 644 /etc/systemd/system/flightairmap.service
	sudo systemctl enable flightairmap.service
	sudo shutdown -r now

	=> run installer in browser http://127.0.0.1/flightairmaps/install

**Comment by TomMuc1** [comment]
GEOID HEIGHT CALCULATOR
all my installs show still this error-message: Exception: Can't open /var/www/html/flightairmap/require/../data/egm96-15.pgm in /var/www/html/flightairmap/require/class.GeoidHeight.php:47 Stack trace: #0 /var/www/html/flightairmap/tools-geoid.php(6): GeoidHeight->__construct() #1 {main}

**Comment by Ysurac** [comment]
Did you run update-db.php ?

**Comment by TomMuc1** [comment]
yep runs every night via cron

**Comment by TomMuc1** [comment]
i'll try a run from terminal as sudo - see what happens then ...

**Comment by TomMuc1** [comment]
when i try to run
    sudo /usr/bin/php /var/www/html/flightairmap/scripts/update_db.php
it says: Script is already runnning... despite i removed from cron and restarted ...

**Comment by Ysurac** [comment]
It's on your mac ? I think it's the check that doesn't work on mac. I check if the script is in the **ps ux** output more than 1 time. If it's the case I will make this check only on Linux.

**Comment by TomMuc1** [comment]
no - it runs on a pi3 with raspian stretch (see above install) only the mysql with flightairmap database runs on the mac-mini server

**Comment by Ysurac** [comment]
I found the problem: sudo put the info 2 times in ps ux. I will try to fix that...

**Comment by TomMuc1** [comment]
great :)

**Comment by Ysurac** [comment]
Fixed in latest commit.

**Comment by TomMuc1** [comment]
solved :) thanx

**Comment by TomMuc1** [comment]
had run update_db.php - took just a few minutes - but the error with geoid remains
GEOID HEIGHT CALCULATOR
all my installs show still this error-message: Exception: Can't open /var/www/html/flightairmap/require/../data/egm96-15.pgm in /var/www/html/flightairmap/require/class.GeoidHeight.php:47 Stack trace: #0 /var/www/html/flightairmap/tools-geoid.php(6): GeoidHeight->__construct() #1 {main}

**Comment by Ysurac** [comment]
What do you have in data directory ?
What the **update_db.php** script say about geoid ?

**Comment by TomMuc1** [comment]
data directory is empty bot has rw from all
closed the terminal already

**Comment by TomMuc1** [comment]
update_db output:

NOTAM are only updated once a day.
DB are populated with external data only every 15 days ! Files are not updated more often.
Owner are only updated every 15 days.
Airlines are only updated every 15 days.
Updating accidents...Accidents are updated once a day.
updating METAR...METAR are only updated every 30 minutes.
Schedules are only updated every 15 days.
Models from FlightAirMap website : Download...Check files...
Done
glTF 2.0 Models from FlightAirMap website : Download...Check files...
Done

**Comment by Ysurac** [comment]
Strange that data are not here.
You have to run this on MySQL:

    delete from config where name ='last_update_geoid' or name ='geoid_version';

Then run update_db.php again.

**Comment by TomMuc1** [comment]
geoidversion was: 49f77c4081021b7b8ff33409fdb99f7b
last update was: 2017-09-29 12:31:32
deleted - now running update_db again will see ...

**Comment by TomMuc1** [comment]
is there something to setup in settings.php that could cause this?

**Comment by Ysurac** [comment]
you should have a file in data dir now

**Comment by TomMuc1** [comment]
no - still empty

**Comment by TomMuc1** [comment]
again data 777 no result -> deleted data -> new data 666 => no result
whats the name of the zip file that should download to data?

NOTAM are only updated once a day.
DB are populated with external data only every 15 days ! Files are not updated more often.
Check if new geoid version exist...Geoid from FlightAirMap website : Download...Gunzip...Done
Owner are only updated every 15 days.
Airlines are only updated every 15 days.
Updating accidents...Accidents are updated once a day.
updating METAR...METAR are only updated every 30 minutes.
Schedules are only updated every 15 days.
Models from FlightAirMap website : Download...Check files...
Done
glTF 2.0 Models from FlightAirMap website : Download...Check files...
Done

**Comment by Ysurac** [comment]
Yes I'm sure and else this would display an error.
Try to git pull latest version and try again, at least you should have more debug info.

**Comment by TomMuc1** [comment]
49f77c4081021b7b8ff33409fdb99f7b
2017-10-02 13:22:02

NOTAM are only updated once a day.
DB are populated with external data only every 15 days ! Files are not updated more often.
Check if new geoid version exist...Geoid from FlightAirMap website : Download...Gunzip...Done
Owner are only updated every 15 days.
Airlines are only updated every 15 days.
Updating accidents...Accidents are updated once a day.
updating METAR...Downloading METAR cycle...Done - Updating DB...Done
Schedules are only updated every 15 days.
Models from FlightAirMap website : Download...Check files...
Done
glTF 2.0 Models from FlightAirMap website : Download...Check files...
Done

**Comment by TomMuc1** [comment]
data dir still empty - can't see more debug

**Comment by TomMuc1** [comment]
see same behavior at another guys install: http://rmm.brainstomp.com:20080/tools/geoid

**Comment by TomMuc1** [comment]
but hey - i can easy life with this bug - no big deal! today showed your apps 3d mode some friends they were really impressed - and i'm too :)

**Comment by Ysurac** [comment]
The error is because the file is not in data dir. Should be here.
Working on my PI. In latest commit I check via PHP if the file exist and display an error if not... No error, the file should exist. No idea here.

**Comment by TomMuc1** [comment]
could you just tell me what script and which line your check for file existence is?

**Comment by daviessm** [comment]
I see the GeoID problem on mine too, didn't notice it until now. I'll try to debug it.

**Comment by Ysurac** [comment]
install/class.update_db.php
function update_geoid_fam
I check if file exist really on line 255

**Comment by TomMuc1** [comment]
line 255 i see just pdo error catching
in update_geoid_fam can't see anything wrong
wonder whether the file is written to another directory???

**Comment by TomMuc1** [comment]
what's the filename i should find in dir data but do not see?

**Comment by Ysurac** [comment]
line 2656 sorry.
egm96-15.pgm.gz is in install/tmp directory but then the gunzip should be data/egm96-15.pgm
I'm downloading lastest raspbian to check if there is no strange settings on PHP cli.

**Comment by TomMuc1** [comment]
no problem - found it via function name you gave me :)
i have php7.0 installed:

pi@RaspPiADSB0:~ $ php -v
PHP 7.0.19-1 (cli) (built: May 11 2017 14:04:47) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.19-1, Copyright (c) 1999-2017, by Zend Technologies

**Comment by TomMuc1** [comment]
while my ftp shows every single file - this one not - but with terminal i see it is there - but still does not work:

pi@RaspPiADSB0:/ $ sudo find . -name '*egm96*' -print
[sudo] password for pi:
./var/www/html/flightairmap/install/tmp/egm96-15.pgm.gz
./var/www/html/flightairmap/install/tmp/egm96-15.pgm.gz.md5
./var/www/html/flightairmap/data/egm96-15.pgm

**Comment by daviessm** [comment]
All I can say is that mine works now. I have no idea why it didn't before and I haven't even done a `git pull` today. After running update_db.php there's egm2008-5.pgm in the data/ directory.

**Comment by TomMuc1** [comment]
thanx for info! i can live with this - enough time spent alltogether on this item with low relevance :)

**Comment by Ysurac** [comment]
This is used to fix altitude of flights in some case (to translate altitude above sea level to above WGS84 ellipsoid), but you can disabled it. In your case this will never be used (ADS-B give GPS altitude that already use WGS84).

**Comment by TomMuc1** [comment]
thanks for explanation and youe time :)

**Comment by TomMuc1** [comment]
finally found the problem - hurray!

if i delete dir data and simply create a new one sudo mkdir /var/www/html/flightairmap/data/
then this new one has 755 permissions and anything works fine - as soon as i reset permissions to 666 it fails again :)

**Comment by Ysurac** [comment]
I added, in install script, a check to know if data is writable or not.

**Comment by TomMuc1** [comment]
yep - and in your howtos - just remove all 666 for data dir as all dirs have normally 755

**Comment by TomMuc1** [comment]
same problem exists for: flightairmap/images/airlines please update howto installs and add the checks in scripts too

cheers
tom

**Comment by TomMuc1** [comment]
another bug or better inconsistency i found is when running install-script and database-writes fail - all inputs are saved despite site location b because this is saved (different from all other parameters) in mysql database and then of course is missing when reloading installscrip.

**Comment by Ysurac** [comment]
wiki is already updated.
Yes it's in database. No choice and better this way.

**Comment by TomMuc1** [comment]
ok :)

**Comment by TomMuc1** [comment]
using the 3d mode of your app on a recent computer is so much fun and just stunning! what I do not understand entirely is why when I see at the moment 150 aircrafts in 2d mode and switch to 3d just about 50 show up it then says 50/150 aircrafts detected but does not increase even within 10 minutes

**Comment by Ysurac** [comment]
In 3D mode it's not really real time because I need more than one coordinate to display a flight correctly. So this will always display less flights than 2D mode.

The 50/150 is the number of flights visible on the screen / the number of flights detected in real time.

Then there is some changes if archive is enabled or not, a limit to 300 3d models displayed at the same time by default,...

**Comment by TomMuc1** [comment]
started a new thread over there at fa: https://discussions.flightaware.com/t/flightairmap-watching-your-live-data-on-a-stunning-new-level/26536/1

**Comment by TomMuc1** [comment]
how much (in seconds) does it leg when you say 'it's not really real time'? this should not or nearly not apply if data-source is aircraft.json

ok - got it

what and where is archive enabeling, does enabling have only influence to this topic in 3d mode or are there other effects/things that rely on this setting?

**Comment by Ysurac** [comment]
* Always less than $globalLiveInterval This apply to all sources. I need to have more than one coordinate to have the flight visible, and I need to have flights to display. No flights send data every seconds.

* Archive is enabled in install/index.php. With this enabled, stats are pre-calculated via update_db.php script (if *generate statistics, [...] older than xx months* > 0). 2D map and 3D map will be faster. When you have many data this make all more responsive.

**Comment by TomMuc1** [comment]
true and not true see below short scan from my site - about 60 seconds from one distinct aircraft

perfect thanx for comprehensive answer! what's the variables name in settings.php i have to look for?

10244	2017-10-03 10:36:59 Tuesday	1507019819.8	06a073	QTR714	36975	50.726944	13.923612	124	523	0	18.8	0.1	-14.4	1112	A0	4741	7		
266	2017-10-03 10:35:56 Tuesday	1507019756.2	06a073	QTR714	37000	50.793147	13.772125	125	523	0	3.5	0	-12.8	754	A0	4741	7		
778	2017-10-03 10:36:00 Tuesday	1507019760.3	06a073	QTR714	37000	50.788119	13.783722	125	523	0	3.8	0.6	-13	779	A0	4741	7		
1291	2017-10-03 10:36:03 Tuesday	1507019763.4	06a073	QTR714	37000	50.778534	13.805732	125	523	0	0	0	-11.7	797	A0	4741	7		
1804	2017-10-03 10:36:06 Tuesday	1507019766.4	06a073	QTR714	36975	50.777971	13.806915	125	523	0	2.6	0.7	-11.1	820	A0	4741	7		
2320	2017-10-03 10:36:09 Tuesday	1507019769.5	06a073	QTR714	37000	50.770294	13.824512	125	523	64	0.1	0.1	-12.3	846	A0	4741	7		
2837	2017-10-03 10:36:12 Tuesday	1507019772.6	06a073	QTR714	37000	50.770294	13.824512	125	523	64	3.2	0	-11.6	853	A0	4741	7		
3356	2017-10-03 10:36:16 Tuesday	1507019776.7	06a073	QTR714	37000	50.761139	13.84552	125	523	64	0.6	0.4	-11	875	A0	4741	7		
3873	2017-10-03 10:36:19 Tuesday	1507019779.8	06a073	QTR714	37000	50.761139	13.84552	125	523	64	3.7	0.8	-10.5	890	A0	4741	7		
4389	2017-10-03 10:36:22 Tuesday	1507019782.9	06a073	QTR714	36975	50.75246	13.86528	125	523	0	0.5	0.1	-9	920	A0	4741	7		
4905	2017-10-03 10:36:26 Tuesday	1507019786	06a073	QTR714	36975	50.748503	13.874435	125	523	0	0.6	0.2	-10.2	943	A0	4741	7		
5422	2017-10-03 10:36:29 Tuesday	1507019789	06a073	QTR714	36975	50.748503	13.874435	125	523	0	3.7	0.3	-10.2	962	A0	4741	7		
5936	2017-10-03 10:36:33 Tuesday	1507019793.1	06a073	QTR714	37000	50.739441	13.895107	125	523	0	1.2	0	-10.1	985	A0	4741	7		
6453	2017-10-03 10:36:36 Tuesday	1507019796.2	06a073	QTR714	37000	50.739441	13.895107	125	523	0	4.3	0	-9.5	1001	A0	4741	7		
6972	2017-10-03 10:36:39 Tuesday	1507019799.3	06a073	QTR714	36975	50.730348	13.915787	125	523	0	0.7	0	-10.3	1017	A0	4741	7		
7491	2017-10-03 10:36:42 Tuesday	1507019802.4	06a073	QTR714	36975	50.726944	13.923612	125	524	0	1.3	0	-11.8	1035	A0	4741	7		
8008	2017-10-03 10:36:45 Tuesday	1507019805.4	06a073	QTR714	36975	50.726944	13.923612	124	523	0	4.4	0	-11.9	1048	A0	4741	7		
8524	2017-10-03 10:36:49 Tuesday	1507019809.5	06a073	QTR714	36975	50.726944	13.923612	124	523	0	8.5	0.3	-10.5	1066	A0	4741	7		
9040	2017-10-03 10:36:52 Tuesday	1507019812.6	06a073	QTR714	36975	50.726944	13.923612	124	523	0	11.6	0.1	-11.2	1082	A0	4741	7		
9555	2017-10-03 10:36:55 Tuesday	1507019815.7	06a073	QTR714	36975	50.726944	13.923612	124	523	0	14.6	0	-12.5	1096	A0	4741	7		
10073	2017-10-03 10:36:58 Tuesday	1507019818.8	06a073	QTR714	36975	50.726944	13.923612	124	523	0	17.7	0.4	-13.6	1106	A0	4741	7		
95	2017-10-03 10:35:55 Tuesday	1507019755.1	06a073	QTR714	37000	50.793147	13.772125	125	523	0	2.5	0.4	-13	750	A0	4741	7		
607	2017-10-03 10:35:58 Tuesday	1507019758.2	06a073	QTR714	37000	50.788119	13.783722	125	523	0	1.7	0.1	-13	776	A0	4741	7		
1120	2017-10-03 10:36:02 Tuesday	1507019762.3	06a073	QTR714	37000	50.788119	13.783722	125	523	0	5.8	0	-13	789	A0	4741	7		
1633	2017-10-03 10:36:05 Tuesday	1507019765.4	06a073	QTR714	36975	50.777971	13.806915	125	523	0	1.5	0	-11	814	A0	4741	7		
2147	2017-10-03 10:36:08 Tuesday	1507019768.5	06a073	QTR714	36975	50.774293	13.815384	125	523	64	1.9	0.2	-11.7	837	A0	4741	7		
2664	2017-10-03 10:36:11 Tuesday	1507019771.6	06a073	QTR714	37000	50.770294	13.824512	125	523	64	2.2	1	-12.3	848	A0	4741	7		
3183	2017-10-03 10:36:14 Tuesday	1507019774.6	06a073	QTR714	37000	50.770294	13.824512	125	523	64	5.2	0.1	-11	861	A0	4741	7		
3701	2017-10-03 10:36:18 Tuesday	1507019778.8	06a073	QTR714	37000	50.761139	13.84552	125	523	64	2.7	0.3	-10.7	889	A0	4741	7		
4217	2017-10-03 10:36:21 Tuesday	1507019781.8	06a073	QTR714	37000	50.761139	13.84552	125	523	0	5.7	0.2	-9.7	907	A0	4741	7		
4733	2017-10-03 10:36:24 Tuesday	1507019784.9	06a073	QTR714	37000	50.750702	13.869348	125	523	0	1.1	0.2	-10.1	939	A0	4741	7		
5251	2017-10-03 10:36:28 Tuesday	1507019788	06a073	QTR714	36975	50.748503	13.874435	125	523	0	2.7	0.2	-10.2	955	A0	4741	7		
5764	2017-10-03 10:36:31 Tuesday	1507019791.1	06a073	QTR714	37000	50.74301	13.886948	125	523	0	1.7	0.3	-10.2	973	A0	4741	7		
6280	2017-10-03 10:36:35 Tuesday	1507019795.2	06a073	QTR714	37000	50.739441	13.895107	125	523	0	3.3	0	-9.4	998	A0	4741	7		
6799	2017-10-03 10:36:38 Tuesday	1507019798.3	06a073	QTR714	37000	50.731605	13.912964	125	523	0	0.6	0.5	-10.2	1012	A0	4741	7		
7318	2017-10-03 10:36:41 Tuesday	1507019801.3	06a073	QTR714	36975	50.726944	13.923612	125	524	0	0.3	0.1	-11.8	1028	A0	4741	7		
7836	2017-10-03 10:36:44 Tuesday	1507019804.4	06a073	QTR714	36975	50.726944	13.923612	124	523	0	3.4	0.3	-11.7	1043	A0	4741	7		
8352	2017-10-03 10:36:47 Tuesday	1507019807.5	06a073	QTR714	36975	50.726944	13.923612	124	523	0	6.4	0.1	-11.3	1057	A0	4741	7		
8868	2017-10-03 10:36:51 Tuesday	1507019811.6	06a073	QTR714	36975	50.726944	13.923612	124	523	0	10.6	0	-11.3	1077	A0	4741	7		
9383	2017-10-03 10:36:54 Tuesday	1507019814.7	06a073	QTR714	36975	50.726944	13.923612	124	523	0	13.6	0.4	-12.2	1092	A0	4741	7		
9900	2017-10-03 10:36:57 Tuesday	1507019817.7	06a073	QTR714	36975	50.726944	13.923612	124	523	0	16.7	1	-13.1	1104	A0	4741	7		
10414	2017-10-03 10:37:00 Tuesday	1507019820.8	06a073	QTR714	36975	50.726944	13.923612	124	523	0	19.8	0	-13.2	1115	A0	4741	7		
437	2017-10-03 10:35:57 Tuesday	1507019757.2	06a073	QTR714	37000	50.788119	13.783722	125	523	0	0.7	0	-12.8	769	A0	4741	7		
949	2017-10-03 10:36:01 Tuesday	1507019761.3	06a073	QTR714	37000	50.788119	13.783722	125	523	0	4.8	0.1	-13.2	786	A0	4741	7		
1462	2017-10-03 10:36:04 Tuesday	1507019764.4	06a073	QTR714	36975	50.777971	13.806915	125	523	0	0.5	0.2	-11.1	806	A0	4741	7		
1975	2017-10-03 10:36:07 Tuesday	1507019767.5	06a073	QTR714	36975	50.774293	13.815384	125	523	64	0.9	0.1	-10.7	827	A0	4741	7		
2492	2017-10-03 10:36:10 Tuesday	1507019770.5	06a073	QTR714	37000	50.770294	13.824512	125	523	64	1.1	0	-12.3	848	A0	4741	7		
3010	2017-10-03 10:36:13 Tuesday	1507019773.6	06a073	QTR714	37000	50.770294	13.824512	125	523	64	4.2	0.1	-11.5	857	A0	4741	7		
3529	2017-10-03 10:36:17 Tuesday	1507019777.7	06a073	QTR714	37000	50.761139	13.84552	125	523	64	1.6	0.1	-11	883	A0	4741	7		
4045	2017-10-03 10:36:20 Tuesday	1507019780.8	06a073	QTR714	37000	50.761139	13.84552	125	523	0	4.7	0	-10.2	901	A0	4741	7		
4561	2017-10-03 10:36:23 Tuesday	1507019783.9	06a073	QTR714	37000	50.750702	13.869348	125	523	0	0.1	0.1	-9.5	926	A0	4741	7		
5078	2017-10-03 10:36:27 Tuesday	1507019787	06a073	QTR714	36975	50.748503	13.874435	125	523	0	1.6	0	-10.4	949	A0	4741	7		
5593	2017-10-03 10:36:30 Tuesday	1507019790.1	06a073	QTR714	37000	50.74301	13.886948	125	523	0	0.7	0.2	-9.6	970	A0	4741	7		
6108	2017-10-03 10:36:34 Tuesday	1507019794.2	06a073	QTR714	37000	50.739441	13.895107	125	523	0	2.3	0	-9.8	991	A0	4741	7		
6626	2017-10-03 10:36:37 Tuesday	1507019797.2	06a073	QTR714	37000	50.732815	13.910141	125	523	0	0.6	0.4	-10.1	1008	A0	4741	7		
7145	2017-10-03 10:36:40 Tuesday	1507019800.3	06a073	QTR714	36975	50.730348	13.915787	125	523	0	1.7	0.1	-11.2	1022	A0	4741	7		
7664	2017-10-03 10:36:43 Tuesday	1507019803.4	06a073	QTR714	36975	50.726944	13.923612	124	523	0	2.4	0.1	-11.6	1038	A0	4741	7		
8180	2017-10-03 10:36:46 Tuesday	1507019806.5	06a073	QTR714	36975	50.726944	13.923612	124	523	0	5.4	0.2	-10.8	1055	A0	4741	7		
8696	2017-10-03 10:36:50 Tuesday	1507019810.6	06a073	QTR714	36975	50.726944	13.923612	124	523	0	9.5	0.1	-11.4	1069	A0	4741	7		
9211	2017-10-03 10:36:53 Tuesday	1507019813.6	06a073	QTR714	36975	50.726944	13.923612	124	523	0	12.6	0	-11.8	1087	A0	4741	7		
9727	2017-10-03 10:36:56 Tuesday	1507019816.7	06a073	QTR714	36975	50.726944	13.923612	124	523	0	15.7	0	-13.1	1104	A0	4741	7

**Comment by Ysurac** [comment]
coordinates are not refreshed each time even if this flights seems to send a lot of messages. Even with this I can't put all data un DB.

$globalArchive and $globalArchiveMonths

**Comment by TomMuc1** [comment]
yes - not every new message holds a new position - just wanted to show that there is a message every second. moreover that is one of the advantages of aircraft.json that you have reliable pre-validated data every single second for all aircrafts in one dump. i did not know that you have to write the data of aircrafts to database before using for map display. i assumed all aircraft data that drives the map just lives in arrays in memory. for transmit rates have a look here at e.g. this eurocontrol document https://www.eurocontrol.int/sites/default/files/publication/files/surveillance-clarification-modes-transponder-in-an-airport-asmgcs-environment-20050403.pdf

cheers
tom

	- Airborne position Squitter
	DF = 17 with GICB register 05 inserted into ME field:
	0,4 to 0,6 second (every half seconds +/- 0,200 seconds)

	- Surface position Squitter
	DF = 17 with GICB register 06 inserted into ME field
	Two cases: High rate/low rate 3
	High rate: 0,4 - 0,6 sec
	Low rate: 4,8 - 5,2 sec

	- Aircraft identification Squitter
	DF = 17 with GICB register 08 inserted into ME field
	￼￼In Air
	On the ground High rate
	Low Rate
	4,8 - 5,2 sec 4,8 - 5,2 sec 9,6 - 10,4 sec

	- Airborne velocity Squitter
	DF = 17 with GICB register 09 inserted into ME field
	Rate 0,4 - 0,6 sec
	- Event Driven Squitter
	DF = 17 with GICB register 0A inserted into ME field
	Limited by the transponder to twice per second

**Comment by Ysurac** [comment]
* I know transmit rate, all aircrafts doesn't send data at the same and all are not received by our receiver. And this doesn't really change anything.
* Aircraft.json format doesn't change anything over any other formats.
* All is in DB. Memory is only used to write less data to DB.

**Comment by TomMuc1** [comment]
ok. what i thought it could change a little was general lag of 3d and how fast aircrafts are displayed when entering 3d mode. why all displayed data has to be written first to db i do not really understand - but i'm sure you have reasons for doing so. anyways i'll now play a little with $globalArchive and $globalArchiveMonths and see what the different outcome there is. thanx as always!

**Comment by Ysurac** [comment]
It's not only a map for dump1090. This can be used in various way. For example if used via cron I can't put anything in memory, spotter_daemon.php can be used on another computer than the webserver,...
Also, sharing memory between PHP processes is not so easy and without problems. In fact it's putting a table in memory, MySQL/MariaDB can also do that if needed.

**Comment by TomMuc1** [comment]
thanx fo answering - always very much appreciated!
i'm fully aware that flightairmap is much more then just a shiny map interface - it is a flightradar24 application out of the box inclusive 2d map, 3d map and a database with lots of possibilties and moreover it can be used for other data than just live aircrafts too. it is a - no question - really great application :)

but it is just as well the by far best map interface available. as lots of the guys out there do not have computers 24/7 running that host a database. the most common machine that runs 24/7 is a raspi with sd-card that is quite slow with database and sd-cards aren't happy with much writing. regretfully even an external usb-harddrive is not really reliable for use with raspi and mysql. there are many guys that will love your map-interface but have a problem to use the database backend of your application. it is already possible to setup the whole thing with lower database usage - but maybe you see the possibility to even lower the the need the database usage for those who mainly want to use the map functionality.

anyways - i'm superhappy with the status quo - and because i'm running the mysql on a 24/7 running mac-mini-server with quadcore and 16 gb ram - all these things mentioned above do not apply to my situation. moreover i love the database backend you deliver!

there is only one single drawback flightairmap has compared to vrs and all the other map apps. but for this item i'll open a different issue within the next days.

**Comment by TomMuc1** [comment]
awesome! played alittle with $globalArchive and $globalArchiveMonths and others. now application is even more responsive and snappy. really unbelivable how fast all menu items appear on screen, how fast maps load, how fast the map is zoom- and draggable. this is so much better than within skyview or virtualradarserver. of course this does not all relate to the new settings - but now anything is superfast.

**Comment by TomMuc1** [comment]
btw found the sweetspot for using the aircraft.json:

$globalLiveInterval = '60'
$globalMinFetch = '1'

**Comment by TomMuc1** [comment]
another thing  - at least my personal preference were is whether the selected map for 2d and 3d could be stored in different variable. that way when switching between both views i did not have to alter the map everytime ...

**Comment by Ysurac** [comment]
I like to keep the same map when I switch between 2d or 3d view, so this would be difficult here. And this was added from a past issue. Maybe adding an option to synchronize map between view or not ?

**Comment by TomMuc1** [comment]
of course this were great - but will this then not overcomplicate things for you :)

**Comment by TomMuc1** [comment]
what i saw in addition is that the number of 3d aircraft models displayed on 3d map extremly influences the computing demands. but i'm not sure what here were the best solution from within user-interface to overcome this. thought about a switch were you could set to show all aircrafts in 3d or just the one that is selected in 2d mode when switching ...

**Comment by Ysurac** [comment]
It's why I limit the number of 3d models to 300 (and this can modified via **$globalMap3DAircraftsLimit**.
I try to optimize aircrafts models but they use lot of memory. 
Chrome/Chromium seems also to be better than firefox on 3D mode.

The switch is a possibility. I will add this.

**Comment by TomMuc1** [comment]
you are simply stunning :) and yes you were fully right to limit this by default and now i learned why. it's not so much here about my needs as we have a small office for product-design there are enough computers that are setup for solidworks and maxwell having enogh cores and reasonable 3d graphic-cards. but those with older equipment or notebooks will be happy if they at all have the chance to use the 3d mode even if it is then only possible with few/selected aircrafts ...

**Comment by TomMuc1** [comment]
installer now complains about different set access rights for dir data and images/airlines since we chmod 755 instead 666

**Comment by Ysurac** [comment]
Installer is right. The test try to write a file to check if it's possible.
chmod 667 should work. Else this would need to be the chown to the user used by PHP or 777.

**Comment by TomMuc1** [comment]
wasn't me - was a guy in fa forum who had this problem ...

**Comment by Ysurac** [comment]
I've added a script to install FlightAirMap and lighttpd on a Rapsberry: https://github.com/Ysurac/FlightAirMap/wiki/Raspbian#script-to-install-using-lighttpd-and-mariadb

I will maybe add one that install also dump1090 & piaware.

**Comment by TomMuc1** [comment]
WOW :) that will make lot of the guys very happy - and i'm sure this will push the user-base of your application again! and i know that this kind of work isn't your (and any coders) biggest love :)))

i'll post the link on the flightaware thread instantly

**Comment by TomMuc1** [comment]
tonight i started just for a short test period a data feed to fr24 and therefore went to their website for the first time since about 2 years. did not know that they now have a 3d mode too that is based on the same code platform you are using for fam. their 3d mode uses much less processing power on the client - and i suppose that is because they use exactly the single-aircraft-mode we were talking about earlier this day. i found no way at their site to use 3d with more than one aircraft (did i overlook sth?) looks like this is really the way to go 2d mode for all aircrafts and 3d (at least default) just for the selected aircraft. what do you think?

**Comment by TomMuc1** [comment]
what i found an advantage there too is that the small map box in the right bottom corner always uses the map view even if the main window shows satellite. this is better to see where you exactly are

**Comment by TomMuc1** [comment]
this runs with ease on my 8 year old macbook pro in safari

![bildschirmfoto 2017-10-04 um 09 33 19](https://user-images.githubusercontent.com/31601862/31164389-3b254f86-a8e7-11e7-9fe0-75b0d8eb8cb5.png)

**Comment by TomMuc1** [comment]
the $globalMap3DAircraftsLimit nowhere exists beside class.SpotterLive.php but i manually added to require/settings.php - now i'll try to lower aircrafts

**Comment by Ysurac** [comment]
* I added an user option to use same 3D model for all aircraft, this will use fewer resources. Displaying one flight use less resources but it's not fun.
* Using another layer for minimap need to load two layers, this will use more resources but can be done.
* I know.

**Comment by TomMuc1** [comment]
- the same model for all aircrafts will show mostly the wrong machine - i do not think that's the way to go
- the fr24 site in 3d mode is super snappy in 3d mode even on my super-old notebook -> so the second layer cannot need too much resources. beside this it is much more intuitive to pick the wanted aircraft in 2d and then watch in 3d. finding the desired aircraft in 3d mode is like tapping in the dark with a stick
- i have no idea whether it is very complicated to code the 'pick in 2d -> display in 3d' mode fr24 has - but no doubt this is the best way by far

**Comment by TomMuc1** [comment]
if it were no fun for the audience to watch single aircrafts in 3d i'm sure fr24 wouldn't go that way ...

**Comment by TomMuc1** [comment]
no question that on superfast machines the show-all-in-3d mode is very cool - but for most users this is simply not possible

**Comment by TomMuc1** [comment]
don't know what the silly tumbs down from the english guy means when i ask to add a second option for those with older machines - anyways ...

**Comment by Ysurac** [comment]
* I think it's better than nothing.
* I will see for second layer.
* I don't want a fr24 clone. This is low priority for me.

**Comment by daviessm** [comment]
Because I agree with @Ysurac - what's the point of making something that's the same as FR24? You could always implement this yourself and submit a pull request to be merged if you really want it.

**Comment by TomMuc1** [comment]
the point is simply to do not leave all those in countries with less money and older computers alone - and help them to use this great app. beside the the guys at fr24 are no idiots what their website prooves. what is your point - what would you miss if others with less money could participate too?

**Comment by TomMuc1** [comment]
- true
- ok
- it's not about cloning more cherry picking - but again if that is very complicated to code i can fully understand that you do not want to spend time in things like this

**Comment by TomMuc1** [comment]
i don't have this problem even on my notebook ...

![screen](https://user-images.githubusercontent.com/31601862/31167973-24daa390-a8f4-11e7-812c-436e4274debd.jpg)

**Comment by Ysurac** [comment]
What is the problem ?
If it's because you see a A320 instead of F100, it's because I don't added Fokker 100 model yet, so it's the default model.

**Comment by TomMuc1** [comment]
:) thanks for reply - but no - there is no problem - i just wanted to show that i do NOT have a problem to watch even larger numbers of aircrafts in 3d even on a tiny notebook.

but i took a look into class.SpotterLive.php and was wondering if there were a sipmle solution by not limiting to max number but a hex-id? if so maybe this could be a simple solution. but i'm noway that deep into all the code of your app and the cesium framework as you of course ...

**Comment by TomMuc1** [comment]
what my problem in general is in 3d mode - i'm not really able to see/find/recognize all or distinct aircrafts. they are very very tiny and often hard to differentiate from satellite background. but no question maybe i'm doing something substantially wrong here?

**Comment by Ysurac** [comment]
ok
I can use an id to show only a flight. But I then need to add setting for that. 
Setting for user or by install ?
I will check into that but not today.

**Comment by Ysurac** [comment]
3D mode is real, flights are difficult to see in real :)
It's why there is an **Force aircraft color** user setting.

**Comment by TomMuc1** [comment]
yes - of course not today. and again - i appreciate all the things you added/altered yet very much! for me it is simply not possible to know what idea/change needs what effort - from my own experience the only thing i know is that of things that look very simple result in a big hassle ...

ah - ok - fine - so at least it is not because i'm to silly to make use of your app :)

**Comment by TomMuc1** [comment]
a user setting in the side-bar would be more convenient for users i think ...

**Comment by TomMuc1** [comment]
was looking where in the cesium files the shadow/shading behavior on the 3d aircraft models is set. did not found - do you know in which files this happens?

**Comment by Ysurac** [comment]
I use CesiumJS but I don't modify it, so no idea.

**Comment by TomMuc1** [comment]
because shadows on my computer look like this - what is different from what i see on cesium tutorial website

![bildschirmfoto 2017-10-04 um 12 48 57](https://user-images.githubusercontent.com/31601862/31172157-c0a50302-a902-11e7-9eec-30ad2bf538a8.png)

**Comment by TomMuc1** [comment]
here you see the standard cesium behavior with shading objects: https://cesiumjs.org/Cesium/Apps/Sandcastle/?src=3D%20Models.html&label=Showcases

![bildschirmfoto 2017-10-04 um 12 53 07](https://user-images.githubusercontent.com/31601862/31172250-1805a6d8-a903-11e7-8a05-1fc4e7e587c6.png)

**Comment by Ysurac** [comment]
This change with altitude, position, model and time of the day.

**Comment by TomMuc1** [comment]
yep - i think too it's a problem of the used 3d aircraft model(s)

**Comment by Ysurac** [comment]
I don't see real problem on your screenshot.

**Comment by TomMuc1** [comment]
when you look at hard edge of the horizontal shade-line on aircraft. don't get this in cesium sample sites or fr24

**Comment by TomMuc1** [comment]
![31172157-c0a50302-a902-11e7-9eec-30ad2bf538a8](https://user-images.githubusercontent.com/31601862/31172739-0eddb35a-a905-11e7-80d2-1f9be3aeaff6.jpg)

**Comment by TomMuc1** [comment]
![bildschirmfoto 2017-10-04 um 13 09 22](https://user-images.githubusercontent.com/31601862/31172936-db8583ec-a905-11e7-9a8c-5262d9d9aa49.png)


![bildschirmfoto 2017-10-04 um 13 13 17](https://user-images.githubusercontent.com/31601862/31172941-e265a1e2-a905-11e7-93c4-7f0129bf26e8.png)

**Comment by Ysurac** [comment]
True not the same : Flightradar24 as no shadow support...

**Comment by TomMuc1** [comment]
cmon :))) of course the have but realistic instead of comic-like. but i don't believe they enhanced something there as the behavior on cesium sites is the same. there is definetly something broken in flightairmaps. probably something on the 3d models site ...

**Comment by Ysurac** [comment]
Flightradar24 don't have sun shadow support. FlightAirMap use sun shadow. It's all.

**Comment by TomMuc1** [comment]
ok - where in the code i can switch this off?

**Comment by Ysurac** [comment]
js/map.3d.js.php -> shadows : true set it to false
This can be a setting.

**Comment by Ysurac** [comment]
But flightradar24 use model shadow. This would need some change to enable only model shadow without sun shadow.

**Comment by TomMuc1** [comment]
haha - thanx :) but it is already much better just without this totally unrrealistic sun-shadow! maybe if it is a quick thing for you change to model shading were a good idea. but i'm happy for now got rid of this sun-shadow-thing

**Comment by TomMuc1** [comment]
but the dafault cesium twin turboprop model in flightairmap seems to have the model shading - just the others not ...

**Comment by TomMuc1** [comment]
here seem to be all fr24 models - no idea whether this helps you or you can make use of: https://github.com/kalmykov/fr24-3d-models

**Comment by Ysurac** [comment]
Flightradar24 use Flightgear models. I use the same models.

**Comment by TomMuc1** [comment]
ok :)

**Comment by TomMuc1** [comment]
git pull -> now get this error in chrome while safari + firefox are fine

	Invalid asm.js: Invalid member of stdlib
	Cesium.js:566 Error constructing CesiumWidget.
	Visit <a href="http://get.webgl.org">http://get.webgl.org</a> to verify that your web browser and hardware support WebGL.  Consider trying a different web browser or updating your video drivers.  Detailed error information is below:
	RuntimeError: The browser supports WebGL, but initialization failed.
	Error
		at new t (Cesium.js:486)
		at new N (Cesium.js:535)
		at new Ee (Cesium.js:552)
		at new A (Cesium.js:565)
		at new Z (Cesium.js:568)
		at map.3d.js.php:281
	A.showErrorPanel @ Cesium.js:566
	Cesium.js:565 Uncaught tmessage: "The browser supports WebGL, but initialization failed."name: "RuntimeError"stack: "Error↵    at new t (https://cesiumjs.org/releases/1.38/Build/Cesium/Cesium.js:486:4061)↵    at new N (https://cesiumjs.org/releases/1.38/Build/Cesium/Cesium.js:535:10114)↵    at new Ee (https://cesiumjs.org/releases/1.38/Build/Cesium/Cesium.js:552:14442)↵    at new A (https://cesiumjs.org/releases/1.38/Build/Cesium/Cesium.js:565:26065)↵    at new Z (https://cesiumjs.org/releases/1.38/Build/Cesium/Cesium.js:568:25781)↵    at http://192.168.143.10/flightairmap/js/map.3d.js.php:281:14"__proto__: Errorconstructor: ƒ t(e)toString: ƒ ()__proto__: Object
	map-aircraft.3d.js.php:537 Uncaught TypeError: Cannot read property 'terrainProvider' of undefined
		at map-aircraft.3d.js.php:537

**Comment by TomMuc1** [comment]
did test the 3d mode extensively in the afternoon. before your changes to master in removed flightairmap and loaded fresh from git. then before running update_db used 3d mode in differnt browsers on different machines. that way only the standard cesium turboprop model was loaded. this changed nearly nothing to the behavior when all different aircraft models are used. then i set $globalMap3DAircraftsLimit = '5' - this time little better at the beginning but after 1-2 minutes it again starts to completely freeze older machines and slow down newer ones. as flightairmap is superfast in 2d mode this is clearly a problem with 3d mode! moreover when using fr24 to test none of the test computers slowed down or nearly freezes.

=> result: there is definitely a bug in the 3d mode of flightairmap as the same cesium framework in fr24 works without any problems.

p.s. fr24 obviously uses sun shading too - as i see the aircraft shadows on ground but never these sharp and unrealistic shadows on their aircraft models

hope this helps a little to identify the bug ...

**Comment by Ysurac** [comment]
* What is the value of cookie *MapTerrain* ?
* For 3D mode all coordinates are used to display the path. If you use a cronjob instead of daemon mode this can be the problem.
* stop comparing to flightradar24... I don't do the same thing at all. We only both use cesiumjs, it's all.
fr24 again don't show sun shadow on 3d models. shadow on 3d model seems realistic for me.

* Some 3d models are bad I know. You can submit pull request to https://github.com/Ysurac/FlightAirMap-3dmodels You only need to know how to use Blender, then export to glTF 2.0 binary format.

**Comment by Ysurac** [comment]
For the error in chrome: https://github.com/AnalyticalGraphicsInc/cesium/issues/5847
Seems to be a chrome 61 problem.

**Comment by Ysurac** [comment]
For 3D mode error in Chrome, issue open here to follow when this will be resolved: https://github.com/Ysurac/FlightAirMap/issues/356

**Comment by TomMuc1** [comment]
thanx - i thought too it were a browser bug since both other browsers work fine.

sorry - but to what should i compare? they use the same framework for the same purpose with the same models. it's not about features they have - they had years of coding hoers, they are a company with a whole team - so no wonder if they might have extended features. i'm just wondering why their app in 3d mode is so much less ressource-hungry than flightairmap. especially as i found fam performance in every single other aspect than 3d overwhelming ...

**Comment by Ysurac** [comment]
They display a model on cesium and I do more than that and not at all the same way used by flightradar24.You don't need to compare to anything.
They can optimize what they want in the way they need, without strange sources or configuration on server side, only one flight,...
Do you fetch the source using cron-job or service ?

**Comment by Ysurac** [comment]
There is also a memory leakage with chrome 61, entities destroy doesn't give memory back.
I will need to make a check for this browser or something...

**Comment by TomMuc1** [comment]
daemon-spotter.php runs as a system service.

good point! you are right controlling the whole server environment makes it a loooot easier that anything runs within the setup they debugged and coded against.

i was sitting additional 2 hoers in front of different monitors, machines and browsers with varying settings of $globalMap3DAircraftsLimit

my imression is really that it does not depend on one model or all models. it only depends on number of aircrafts in 3d mode

if the fam standard with 3d mode were the same fr24 uses - means: select aircraft in 2d -> chose 3d -> display this single aircraft - anything were fine and without any problem

the problem simply is to have more objects in parallel in 3d. this is a very nice option for those with really fast machines - but as default not a good idea in my opinion

p.s. yes - funny thing is that install with safari is a catastrophe but using fam i'm more happy with safari than chrome. and this is nive as safari is my all day browser :)

**Comment by TomMuc1** [comment]
p.s. nooooo - dont start this browser-check-horror!!! this is a never ending story and every next day you can start to adapt new check-trules ...

it is what it is - and over time when the fam user base is bigger there will be a forum and user can help users without annoying you with every little hickup here and there

**Comment by Ysurac** [comment]
When I speak about browser check, it's only a check in install script that tell "Warning, Chrome 61 is buggy as hell" :)

**Comment by TomMuc1** [comment]
ok - this kind of browser check wont bind too much of your resources :)))

**Comment by TomMuc1** [comment]
i'd say this issue has done it's job and got quite extensive :) so if i find other little bugs or whatever i'll open part 2
